src/main/java/com/internet/shop/dao/impl/UserDaoImpl.java

```diff
@@ -42,4 +42,11 @@ public boolean delete(Long id) {
         return Storage.userStorage
                 .removeIf(user -> user.getId().equals(id));
     }
+
+    @Override
+    public Optional<User> findByLogin(String login) {
+        return Storage.userStorage.stream()
+                .filter(user -> user.getLogin().equalsIgnoreCase(login))
```

Comment:
```suggestion
                .filter(user -> user.getLogin().equals(login))
```
login should be 100% the same